extension/src/openvic-extension/singletons/CursorSingleton.hpp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <godot_cpp/core/class_db.hpp>
+#include <godot_cpp/core/object.hpp>
+#include <godot_cpp/classes/file_access.hpp>
+#include <godot_cpp/classes/image_texture.hpp>
+#include <godot_cpp/classes/input.hpp>
+#include <godot_cpp/templates/vector.hpp>
+#include <godot_cpp/variant/packed_string_array.hpp>
+#include <godot_cpp/variant/string.hpp>
+#include <godot_cpp/variant/string_name.hpp>
+#include <godot_cpp/variant/typed_array.hpp>
+
+#include <openvic-extension/singletons/GameSingleton.hpp>
+#include <openvic-simulation/dataloader/Dataloader.hpp>
+#include <openvic-simulation/types/OrderedContainers.hpp>
+#include <openvic-simulation/types/IdentifierRegistry.hpp>
+
+namespace OpenVic {
+
+	class CursorSingleton : public godot::Object {
+		
+		/*
+		This singleton only handles the dataloading for the cursors.
+		The functionality for actually setting one of these as a cursor
+		and animating animated cursors is done by the CursorManager autoloaded
+		gd script.
+		*/
```

Comment:
Documentation should be in `CursorSingleton.xml` following https://docs.godotengine.org/en/latest/contributing/documentation/index.html